examples/js/plugins/COGParser.js

```diff
@@ -0,0 +1,306 @@
+            let g = gChannel[i];
+            let b = bChannel[i];
+
+            if ((r !== r || r === options.noData)
+                && (g !== g || g === options.noData)
+                && (b !== b || b === options.noData)) {
+                r = source.defaultNoData;
+                g = source.defaultNoData;
+                b = source.defaultNoData;
+                a = source.defaultTransparent;
+            } else {
+                r = getValue(r);
+                g = getValue(g);
+                b = getValue(b);
+                a = opaqueValue;
+            }
+
+            buffers[idx + 0] = r;
+            buffers[idx + 1] = g;
+            buffers[idx + 2] = b;
+            buffers[idx + 3] = a;
+        }
+    }
+    if (pixelData.length === 4) {
+        const rChannel = pixelData[0];
+        const gChannel = pixelData[1];
+        const bChannel = pixelData[2];
+        const aChannel = pixelData[3];
+        const length = rChannel.length;
+        for (let i = 0; i < length; i++) {
+            const idx = i * 4;
+            let r = rChannel[i];
+            let g = gChannel[i];
+            let b = bChannel[i];
+            let a = aChannel[i];
+
+            if ((r !== r || r === options.noData)
+                && (g !== g || g === options.noData)
+                && (b !== b || b === options.noData)) {
+                r = source.defaultNoData;
+                g = source.defaultNoData;
+                b = source.defaultNoData;
+                a = source.defaultTransparent;
+            } else {
+                r = getValue(r);
+                g = getValue(g);
+                b = getValue(b);
+            }
+
+            buffers[idx + 0] = r;
+            buffers[idx + 1] = g;
+            buffers[idx + 2] = b;
+            buffers[idx + 3] = a;
+        }
+    }
+    return buffers;
+}
+
+/**
+ * The COGParser module provides a [parse]{@link module:COGParser.parse}
+ * method that takes a COG in and gives a `THREE.DataTexture` that can be
+ * displayed in the view.
+ *
+ * It needs the [geotiff](https://github.com/geotiffjs/geotiff.js/) library to parse the
+ * COG.
+ *
+ * @example
+ * GeoTIFF.fromUrl('http://image.tif')
+ *     .then(COGParser.parse)
+ *     .then(function _(texture) {
+ *         var source = new itowns.FileSource({ features: texture });
+ *         var layer = new itowns.ColorLayer('cog', { source });
+ *         view.addLayer(layer);
+ *     });
+ *
+ * @module COGParser
+ */
+const COGParser = (function _() {
+    if (typeof THREE == 'undefined'  && itowns.THREE) {
+        // eslint-disable-next-line no-global-assign
+        THREE = itowns.THREE;
+    }
+
+    return {
+        /**
+         * Parse a COG file and return a `THREE.DataTexture`.
+         *
+         * @param {Object} data Data passed with the Tile extent
+         * @param {Object} options Options (contains source)
+         * @return {Promise} A promise resolving with a `THREE.DataTexture`.
```

Comment:
Missing types of parameters:

```suggestion
         * @param {Object} data Data passed with the Tile extent
         * @param {Extent} data.extent
         * @param {Object} options Options (contains source)
         * @param {Object} options.in
         * @param {COGSource} options.in.source
         * @param {number} options.in.tileWidth
         * @param {number} options.in.tileHeight
         * @return {Promise<THREE.DataTexture>} A promise resolving with a `THREE.DataTexture`.
```

Reply:
Ok, done